tests/components/admin_console/manage_teams_modal/manage_teams_modal.test.jsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import ManageTeamsModal from 'components/admin_console/manage_teams_modal/manage_teams_modal.jsx';
+
+describe('ManageTeamsModal', () => {
+    function emptyFunc() {} // eslint-disable-line no-empty-function
+
+    const baseProps = {
+        onModalDismissed: emptyFunc,
+        show: true,
+        user: {
+            id: 'currentUserId',
+            last_picture_update: '1234',
+            email: 'currentUser@test.com',
+            roles: 'system_user',
+            username: 'currentUsername',
+        },
+        updateTeamMemberSchemeRoles: emptyFunc,
+        getTeamMembersForUser: jest.fn(),
+        getTeamsForUser: jest.fn(),
+    };
+
+    test('should match snapshot init', () => {
+        const wrapper = shallow(
+            <ManageTeamsModal {...baseProps}/>
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should call api calls on mount', () => {
+        shallow(
+            <ManageTeamsModal {...baseProps}/>
+        );
+
+        expect(baseProps.getTeamMembersForUser).toHaveBeenCalledTimes(1);
+        expect(baseProps.getTeamMembersForUser).toHaveBeenCalledWith('currentUserId');
+        expect(baseProps.getTeamsForUser).toHaveBeenCalledTimes(1);
+        expect(baseProps.getTeamsForUser).toHaveBeenCalledWith('currentUserId');
+    });
+
+    test('should save data in state from api calls', async () => {
+        const mockTeamData = {
+            id: '123test',
+            name: 'testTeam',
+            display_name: 'testTeam',
+            delete_at: 0,
+        };
+
+        const getTeamMembersForUser = jest.fn(() => {
+            return Promise.resolve({
+                data: [{team_id: '123test'}],
+            });
+        });
+
+        const getTeamsForUser = jest.fn(() => {
+            return Promise.resolve({data: [mockTeamData]});
+        });
+
+        const wrapper = shallow(
+            <ManageTeamsModal
+                {...baseProps}
+                getTeamMembersForUser={getTeamMembersForUser}
+                getTeamsForUser={getTeamsForUser}
+            />
+        );
+
+        getTeamsForUser();
+        await getTeamMembersForUser();
```

Comment:
And this line doesn't do anything directly, but it indirectly forces other async calls to resolve. I'd prefer to use something explicit like `process.nextTick` here since it's clear what it's doing as opposed to the current version that waits on a Promise to force the async calls to resolve.

Reply:
I updated the tests and had to do a bunch of changes as master has changed since last time. 

`process.nextTick` is equivalent to  setTimeout and that does not help me understand about the nature of processes we are waiting in the test case. Makes me think it is as brittle. Anyway probably just my personal opinion. I changed that test to have `process.nextTick` but i still needed to keep async for snapshot or otherwise it does not get generated.

Reply:
I had to remove process tick as even though the tests pass snapshot does not get generated with it. I would like to have a snapshot for the updated render tree.

Feel free to any variation you see fit and suggest

Reply:
I think you'd need to call done explicitly if you do that like
```javascript
test('something', (done) => {
    // do an action

    process.nextTick(() => {
        // assert stuff
        done();
    });
});
```

Reply:
Made changes let me know if this works. 
I am more inclined to creating a ticket for the test case here so we can iterate on it separately and does not block the PR for other changes